lib/manager/nuget/extract.ts

```diff
@@ -1,19 +1,65 @@
+import findUp from 'find-up';
+import * as path from 'path';
+import { XmlDocument } from 'xmldoc';
+import { readFile } from 'fs-extra';
 import { logger } from '../../logger';
 import { get } from '../../versioning';
 import { PackageDependency, ExtractConfig, PackageFile } from '../common';
 import * as semverVersioning from '../../versioning/semver';
 import * as datasourceNuget from '../../datasource/nuget';
 import { SkipReason } from '../../types';
 
-export function extractPackageFile(
+async function determineRegistryUrls(
+  packageFile: string,
+  localDir: string
+): Promise<string[]> {
+  const nuGetConfigPath = await findUp('NuGet.config', {
+    cwd: path.dirname(path.join(localDir, packageFile)),
+    type: 'file',
+  });
+  if (nuGetConfigPath && nuGetConfigPath.startsWith(localDir)) {
```

Comment:
We should check for the negative here and return undefined if so. Logically simpler and saves whitespace below.

Reply:
```suggestion
  if (nuGetConfigPath?.startsWith(localDir) !== true) {
      return undefined;
  }
```

Reply:
Done in 24a4d5d806ca2ee639d2e9c7a0ed6fc5066584bc